openhcl/underhill_attestation/src/igvm_attest/key_release.rs

```diff
@@ -162,14 +170,54 @@ pub fn parse_response(
     Ok(wrapped_key)
 }
 
+/// Convert a potentially non UTF-8 byte array into a string with non UTF-8 characters represented
+/// as hexadecimal escape sequences.
+fn string_from_utf8_preserve_invalid_bytes<T: AsRef<[u8]>>(bytes: T) -> String {
```

Comment:
rather than returning a new string each time, have the recursive part of this accept a `&mut String` that can be extended into

Reply:
Addressed in the latest implementation (https://github.com/microsoft/openvmm/pull/583/commits/e3156eeddb24a549250d5e534635d881b456f4f4)